src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -442,16 +443,15 @@ private static void ReportCA1836(OperationAnalysisContext context, string operat
                     else
                     {
                         if (allowedTypesForCA1836.Contains(type.OriginalDefinition) &&
-                            TypeContainsVisibleProperty(context, type, IsEmpty, SpecialType.System_Boolean, out ISymbol? isEmptyPropertySymbol) &&
-                            !IsPropertyGetOfIsEmptyUsingThisInstance(context, invocationOperation, isEmptyPropertySymbol!))
+                            TypeContainsVisiblePropertyWithoutCircularity(context, invocationOperation, type, IsEmpty, SpecialType.System_Boolean, out _))
                         {
                             ReportCA1836(context, operationKey!, shouldNegateIsEmpty, parent);
                         }
-                        else if (TypeContainsVisibleProperty(context, type, Length, SpecialType.System_Int32, SpecialType.System_UInt64, out _))
+                        else if (TypeContainsVisiblePropertyWithoutCircularity(context, invocationOperation, type, Length, SpecialType.System_Int32, SpecialType.System_UInt64, out _))
                         {
                             ReportCA1829(context, Length, invocationOperation);
                         }
-                        else if (TypeContainsVisibleProperty(context, type, Count, SpecialType.System_Int32, SpecialType.System_UInt64, out _))
+                        else if (TypeContainsVisiblePropertyWithoutCircularity(context, invocationOperation, type, Count, SpecialType.System_Int32, SpecialType.System_UInt64, out _))
```

Comment:
Can this fall to the `else` clause and report CA1829 when it shouldn't? I haven't read the whole implementation.

Reply:
I was reading your changes and the code and I am not entierly sure to get to what you are referring to here, do you remember?

Reply:
@Evangelink I expanded the "TypeContainsVisibleProperty" method to also check for circularity. See https://github.com/dotnet/roslyn-analyzers/commit/41c172b8d32a5f459bedf705dac21cdb2bae2f13#diff-2f73c823f9e61fa28f679e58facef5ce89e1385f75d725815fb6f453dd2dc3c2R411-R412 where this was originally fixed, but didn't cover everything. Instead of checking in the caller-side, I moved the check inside the method.